wireapp/wire-ios-mono
# fix:  Image selector anchor - WPB-11605

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11605" title="WPB-11605" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-11605</a>  [iOS] Anchor profile image selector popover
  </td></table>
  <br />
 





### Issue

Currently on the navigation overhaul epic branch the popover `UIImagePickerController` is not properly anchored as it doesn't have a suitable `sourceView` set.

This PR:

- Removes an unreferenced protocol, `SettingsExternalScreenCellDescriptorType`, allowing more flexibility when creating `SettingsAppearanceCellDescriptor`, and thus allowing the `sender` of the settings screen to be forwarded to the `UIImagePickerController`s picker.
- Updates `ImagePickerManager.showActionSheet(on:completion:)` to accept a `popoverSourceView` which gets set on the popover controller.
- Uses a popover controller when displaying `UIImagePickerController` on any iPad, not just regular size classes as is necessary for apples API.
- Does some general clean up to simplify code and remove optional instance variables

### Testing

0. Please test on a **real** iPad.
1. Navigate to `Setting`
2. Tap `Profile Picture`
3. Tap `Choose from Library`
4. Photo picker should be well placed
5. Navigate to profile
6. Tap on your profile picker
7. Tap `Choose from Library`
8. Photo picker should be well placed

Also try rotating the device to check things are working

#### NOTE:

After selecting a photo there is a know issue that is not fixed in this PR: https://wearezeta.atlassian.net/browse/WPB-11929

### Checklist
